src/main/java/com/hackclub/hccore/advancements/MusicophileAdv.java

```diff
@@ -0,0 +1,79 @@
+package com.hackclub.hccore.advancements;
+
+import com.fren_gor.ultimateAdvancementAPI.advancement.Advancement;
+import com.fren_gor.ultimateAdvancementAPI.advancement.BaseAdvancement;
+import com.fren_gor.ultimateAdvancementAPI.advancement.display.AdvancementDisplay;
+import com.fren_gor.ultimateAdvancementAPI.advancement.display.AdvancementDisplayBuilder;
+import com.fren_gor.ultimateAdvancementAPI.util.AdvancementKey;
+import com.fren_gor.ultimateAdvancementAPI.util.CoordAdapter;
+import com.hackclub.hccore.HCCorePlugin;
+import org.bukkit.Material;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.entity.EntityPickupItemEvent;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.metadata.FixedMetadataValue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MusicophileAdv extends BaseAdvancement {
```

Comment:
this implementation seems odd to me:
it adds metadata to the player

Reply:
Maybe there's a better way to do this (I'm new to mc plugins and don't have much experience with java in general), but I wasn't sure how to track which disks the player had already collected. This seemed like the way to do it.

Reply:
ill be honest, this api doesnt seem to support individual criterion tracking, only an int to represent progress :/
in a vanilla datapack, the data is held within the advancement itself
even bukkit itself: Player#getAdvancementProgress .awardCriteria
(or similar)

Reply:
this is probably fine, if we change implementations later, then it can either be cleaned up in the replacement, or somthing else